index.bs

```diff
@@ -159,29 +161,25 @@ Resource names starting with U+002D HYPHEN-MINUS (-) are reserved; requesting th
 ## Lock Managers ## {#lock-managers}
 <!-- ====================================================================== -->
 
-A user agent has a <dfn>lock manager</dfn> for each [=/origin=], which encapsulates the state of all [=lock-concept|locks=] and [=lock requests=] for that origin.
+A [=/user agent=] has a <dfn>lock manager</dfn> for each [=/storage bottle=], which encapsulates the state of all [=lock-concept|locks=] and [=lock requests=] associated with a [=/storage bucket=].
 
-Pages and workers ([=/agents=]) on a single [=/origin=] opened in the same user agent share a lock manager even if they are in unrelated [=/browsing contexts=].
+Note: Pages and workers ([=/agents=]) sharing a [=/storage bucket=] opened in the same user agent share a [=/lock manager=] even if they are in unrelated [=/browsing contexts=].
 
-<aside class=note>
-
-    There is an equivalence between the following:
-
-    * Agents that share a [=lock manager=].
-    * Agents that share a [=/storage shelf=], i.e. a per-origin [=localStorage|local storage area=] [[HTML]], set of [[IndexedDB-2#database-construct|databases]] [[IndexedDB-2]], or [[Service-Workers#cache-objects|caches]] [[Service-Workers]].
+<div algorithm>
+To <dfn>obtain a lock manager</dfn>, given an [=/environment settings object=] |environment|, run these steps:
 
-</aside>
+1. Let |bottle| be the result of [=/obtaining a local storage bottle map=] given |environment| and "`web-locks`".
+1. If |bottle| is failure, then return failure.
+1. Return |bottle|'s associated [=/lock manager=].
```

Comment:
The way the Storage standard is written and algorithms exposed, without changes to the standard "client" specs like this can get only a bottle for a feature. I think the intent is that for something like Indexed DB the bottle would literally be the key/value map where data is stored - for IDB, presumably a map of names to databases. 

In the case of locks here, there's a top level "manager" object. With the bottle construct, we could:
* Have the manager live inside the bottle map, as the value of a single key
* Rewrite the manager to store all of its state as key/value pairs within the bottle map.
* Monkey-patch the bottle map to *be* the manager with additional state properties.
* "Associate" a manager with the specific bottle

The last is done here, basically bolting on a lock manager to a bottle, which is not very formal. (And again, it used to say "bucket" here) The text here is again the result of a refactor where this used to say "bucket", but since buckets aren't exported from Storage, that doesn't work. 

I don't have a strong preference, but I do find the "store all state as key/value pairs within the bottle map" to be somewhat awkward/constraining as a spec author. I would prefer that specs can be written without modifications to Storage, i.e. we do keep the correct layering. 

Have any other specs adopted the bottle mechanism? I put up an early PR for IDB but haven't looked at it in a while. With no changes to bottle, my preference would probably be to have keep the lock manager as-is, as a single value within the bottle. It seems like overkill, but spec objects are cheap. :)

Reply:
HTML has for `localStorage`/`sessionStorage`. It might be more straightforward there.

But the idea is that nobody gets to access a bottle or bucket directly as they are in the conceptual "storage process". APIs just get a proxy to some space within that process.

Reply:
Hmm, that helps point out the challenge. Things like the lock manager and IDB database connection queues and transaction manager conceptually run algorithms in the "storage process" itself, so we need specs to have more than just proxied access to a k/v store.

Reply:
Even if we keep this abstract for now, it seems wrong that above we associate it with a bucket, but here we pretend it's associated with a bottle.